### ISSUE TITLE
Avoid panics in the sentinel

### ISSUE THREAD

**Issue by Forsworns** [sentinel-group/sentinel-rust#8]
We should try our best to eliminate panics in the `sentinel` library. The errors should be propagated to the user or written into logs, instead of simply generating a fatal panic silently.

**Comment by Millione** [comment]
Recently, I got a fatal panic from https://github.com/sentinel-group/sentinel-rust/blob/main/sentinel-core/src/core/stat/base/bucket_leap_array.rs#L18. I wonder if I could just use tracing::error!() instead of panic?

**Comment by Forsworns** [comment]
@Millione Yes sentinel should avoid these `unwrap` by throwing out the errors to the API caller. But currently it just panic silently. You can submit a PR if you are interested, thanks :)
